src/main/java/com/rultor/agents/github/CommentsTag.java

```diff
@@ -94,44 +106,191 @@ public CommentsTag(final Github ghub) {
         }
         return new Directives();
     }
 
     /**
+     * Create the release and add a comment.
+     * @param issue The issue to add a comment to.
+     * @param tag The tag for the release.
+     * @param rels The existing releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
+     */
+    private void release(final Issue.Smart issue, final String tag,
+        final Releases rels, final URI home) throws IOException {
+        final Repo repo = issue.repo();
+        final Date prev = CommentsTag.previous(repo);
+        final Release.Smart rel = new Release.Smart(rels.create(tag.trim()));
+        rel.name(issue.title());
+        rel.prerelease(true);
+        rel.body(
+            String.format(
+                // @checkstyle LineLength (1 line)
+                "See #%d, release log:\n\n%s\n\nReleased by Rultor %s, see [build log](%s)",
+                issue.number(),
+                new CommitsLog(repo).build(prev, rel.publishedAt()),
+                Manifests.read("Rultor-Version"), home
+            )
+        );
+        Logger.info(this, "tag %s created and commented", tag);
+    }
+
+    /**
+     * Logs an issue because the new tag is too old.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that for the new release.
+     * @param previous The previous releases.
+     * @throws IOException If there is a problem.
+     */
+    private void logTagTooOld(final Issue issue, final String tag,
+        final List<DefaultArtifactVersion> previous) throws IOException {
+        issue.comments().post(
+            String.format(
+                // @checkstyle LineLength (1 line)
+                CommentsTag.PHRASES.getString("CommentsTag.version-too-low"),
+                tag, previous.toString()
+            )
+        );
+        Logger.info(
+            this, "tag %s must be greater than previous version %s",
+            tag, previous
+        );
+    }
+
+    /**
+     * Logs a comment because the tag is invalid.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag which is invalid.
+     * @throws IOException If there is a problem.
+     */
+    private void logInvalidTag(final Issue issue, final String tag)
+        throws IOException {
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.invalid-tag"), tag
+            )
+        );
+        Logger.info(this, "tag %s is invalid", tag);
+    }
+
+    /**
+     * Logs a comment because the tag has already been released.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that has already been released.
+     * @param rels The other releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
```

Comment:
@gumbelmj instead of suppressing the check that is supposed to indicate to you that your method is too complex by using this many inputs, you need to create a method in compliance with the check.